inc/saivlan.h

```diff
@@ -98,6 +98,12 @@ typedef enum _sai_vlan_attr_t
       * Value Range SAI_SWITCH_ATTR_VLAN_USER_META_DATA_RANGE */
     SAI_VLAN_ATTR_META_DATA,
 
+    /**
+      * The Multicast 802.1BR E-Channel Id (ECID), to be used in the flooding traffic
+      * sent by the CB to Port Extenders (PE). [sai_uint32_t] (CREATE_AND_SET) (default to 0)
+      */
+    SAI_VLAN_ATTR_FLOODING_ECID,
```

Comment:
Dell - Internal Use - Confidential
Hi Min Yao,

You are right, the Standard says that the ECID must be unique with an Internal Bridge Port Extender.

For each Cascading Port in a CB, there is an associated, Internal Bridge Port Extender. Each Extended Port (remote port on the external PE) has an associated Internal Bridge Port. As per the standard, an ECID assigned to the External Extended Port can also be associated to another Extended Port, as long as these External Extended Ports are not associated to the same Internal Bridge Port Extender (or in other words, these external extended ports are not reachable via same Cascading Port).

But when a ECID is assigned to a Vlan for flooding, in a global scope, it will still not violate the Standards’ requirement of having unique ECID with in an Internal Bridge Port Extender. A CB is a system, containing the Internal Bridge Port Extenders as its subsystem. If the ECID is unique in CB, it would still be unique within each of its subsystems (Internal Bridge Port Extenders).

If we were to do otherwise, then each Vlan must be associated with a list of {Cascading Port, ECID} tuple. This would really complicate the SAI functionality.
Regards,
Ravi

From: Min Yao [mailto:notifications@github.com]
Sent: Tuesday, August 30, 2016 8:08 PM
To: opencomputeproject/SAI <SAI@noreply.github.com>
Cc: Sivasankar, Ravikumar <Ravikumar_Sivasankar@Dell.com>; Mention <mention@noreply.github.com>
Subject: Re: [opencomputeproject/SAI] IEEE 802.1BR SAI proposal. (#73)


In inc/saivlan.h<https://github.com/opencomputeproject/SAI/pull/73#discussion_r76807888>:

> @@ -98,6 +98,12 @@ typedef enum _sai_vlan_attr_t

>        * Value Range SAI_SWITCH_ATTR_VLAN_USER_META_DATA_RANGE */

>      SAI_VLAN_ATTR_META_DATA,

>

> +    /**

> +      * The Multicast 802.1BR E-Channel Id (ECID), to be used in the flooding traffic

> +      * sent by the CB to Port Extenders (PE). [sai_uint32_t] (CREATE_AND_SET) (default to 0)

> +      */

> +    SAI_VLAN_ATTR_FLOODING_ECID,

@rsivasankar<https://github.com/rsivasankar> Sorry, mistype. “Internal External Bridge” -->> "Internal Bridge Port Extender".

An ECID is allocated within the scope the an "Internal Bridge Port Extender". Check IEEE 802.1br page 40, Figure 8-3,

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/opencomputeproject/SAI/pull/73/files/279f0335c33c0c831ec955e1cf93daf61adf52d5#r76807888>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AOoHu6BTA78kRuCrO931w7nlcB1FSszqks5qlEBGgaJpZM4HpSGE>.